frontend/src/store/NotificationStore.ts

```diff
@@ -0,0 +1,94 @@
+import { defineStore } from 'pinia';
+import { reactive, ref, watchEffect } from 'vue';
+import { v4 as uuidv4 } from 'uuid';
+import duration from 'dayjs/plugin/duration';
+
+export enum NotificationType {
+  SUCCESS = 'success',
+  WARNING = 'warning',
+  ERROR = 'error',
+  PROGRESS = 'progress',
+}
+
+export interface Notification {
+  id: string;
+  type: NotificationType;
+  message: string;
+  closable?: true;
+  progress?: number;
+}
+
+export const useNotificationStore = defineStore('notification', () => {
+  const notifications = reactive<Record<string, Notification>>({});
+
+  const remove = (id: string) => {
+    delete notifications[id];
+  };
+
+  const create = (type: NotificationType, message: string, progress?: number): string => {
```

Comment:
It make probably more sense to accept an object directly to add flexibility to the params 

```show = ({type: NotificationType, message: string, progress?: number}): string => {



```

Reply:
Did something similar. Hope you like it